File: src/DashboardUI/src/data-contracts/WaterRightsSearchCriteria.ts
export interface WaterRightsSearchCriteria {
  beneficialUses?: string[];
  ownerClassifications?: string[];
  waterSourceTypes?: string[];
  riverBasinNames?: string[];
  wadeSitesUuids?: string[];
  filterGeometry?: string[]; //geojson string
  states?: string[];
  allocationOwner?: string;
  exemptofVolumeFlowPriority?: boolean;
  minimumFlow?: number;
  maximumFlow?: number;
  minimumVolume?: number;
  maximumVolume?: number;
  podOrPou?: string;
  minimumPriorityDate?: Date;
  maximumPriorityDate?: Date;
}

export interface WaterRightsSearchCriteriaWithPaging extends WaterRightsSearchCriteria {
  pageNumber: number;
}

export interface WaterRightsSearchCriteriaWithFilterUrl extends WaterRightsSearchCriteria {
  filterUrl: string;
}

export interface WaterRightsSearchCriteriaWithGrouping extends WaterRightsSearchCriteria {
  groupValue?: number;
}

File: src/API/WesternStatesWater.WestDaat.Contracts.Client/WaterRightsSearchCriteria.cs
namespace WesternStatesWater.WestDaat.Contracts.Client
{
    public class WaterRightsSearchCriteria
    {
        public string[] BeneficialUses { get; set; }
        public string[] OwnerClassifications { get; set; }
        public string[] WaterSourceTypes { get; set; }
        public string[] RiverBasinNames { get; set; }
        public string[] WadeSitesUuids { get; set; }

        // Geojson string
        public string[] FilterGeometry { get; set; }

        public string[] States { get; set; }
        public string AllocationOwner { get; set; }
        public bool? ExemptOfVolumeFlowPriority { get; set; }
        public double? MinimumFlow { get; set; }
        public double? MaximumFlow { get; set; }
        public double? MinimumVolume { get; set; }
        public double? MaximumVolume { get; set; }
        public string PodOrPou { get; set; }
        public DateTime? MinimumPriorityDate { get; set; }
        public DateTime? MaximumPriorityDate { get; set; }
    }
}
